cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseVersion.java

```diff
@@ -235,43 +235,55 @@ private static void determineVersion() {
   static Version determineVersionFromVersionString(String versionString) throws ParseException {
     if (versionString.startsWith(HBASE_94_VERSION)) {
       return Version.HBASE_94;
-    } else if (versionString.startsWith(HBASE_96_VERSION)) {
+    }
+    if (versionString.startsWith(HBASE_96_VERSION)) {
       return Version.HBASE_96;
-    } else if (versionString.startsWith(HBASE_98_VERSION)) {
+    }
+    if (versionString.startsWith(HBASE_98_VERSION)) {
       return Version.HBASE_98;
-    } else if (versionString.startsWith(HBASE_10_VERSION)) {
-      VersionNumber ver = VersionNumber.create(versionString);
-      if (ver.getClassifier() != null && ver.getClassifier().startsWith(CDH55_CLASSIFIER)) {
-        return Version.HBASE_10_CDH55;
-      } else if (ver.getClassifier() != null && ver.getClassifier().startsWith(CDH56_CLASSIFIER)) {
-        return Version.HBASE_10_CDH56;
-      } else if (ver.getClassifier() != null && ver.getClassifier().startsWith(CDH_CLASSIFIER)) {
-        return Version.HBASE_10_CDH;
-      } else {
-        return Version.HBASE_10;
-      }
-    } else if (versionString.startsWith(HBASE_11_VERSION)) {
+    }
+    VersionNumber ver = VersionNumber.create(versionString);
+    if (versionString.startsWith(HBASE_10_VERSION)) {
+      return getHBase10VersionFromVersionString(ver);
+    }
+    if (versionString.startsWith(HBASE_11_VERSION)) {
       return Version.HBASE_11;
-    } else if (versionString.startsWith(HBASE_12_VERSION)) {
-      VersionNumber ver = VersionNumber.create(versionString);
-      if (ver.getClassifier() != null &&
-        (ver.getClassifier().startsWith(CDH57_CLASSIFIER) ||
-          // CDH 5.7 compat module can be re-used with CDH 5.[8-11].x
-          ver.getClassifier().startsWith(CDH58_CLASSIFIER) ||
-          ver.getClassifier().startsWith(CDH59_CLASSIFIER) ||
-          ver.getClassifier().startsWith(CDH510_CLASSIFIER) ||
-          ver.getClassifier().startsWith(CDH511_CLASSIFIER))) {
+    }
+    if (versionString.startsWith(HBASE_12_VERSION)) {
+      return getHBase12VersionFromVersionString(ver);
+    }
+    if (ver.getClassifier() != null && ver.getClassifier().startsWith(CDH_CLASSIFIER)) {
+      return Version.UNKNOWN_CDH;
+    }
+    return Version.UNKNOWN;
+  }
+
+  private static Version getHBase10VersionFromVersionString(VersionNumber ver) {
```

Comment:
`getHBase10VersionFromVersionString` is actually `getHBase10VersionFromVersion`, right?